packit/api.py

```diff
@@ -1490,42 +1490,47 @@ def _prepare_files_to_sync(
         resolved_bugs: Optional[list[str]] = None,
     ) -> list[SyncFilesItem]:
         """
-        Returns the list of files to sync to dist-git as is.
+        Returns the list of files to sync to dist-git with the spec file removed.
 
-        With spec-file, we have two modes:
-        * The `sync_changelog` option is set. => Spec-file can be synced as other files.
-        * Sync the content of the spec-file (but changelog) here and exclude spec-file otherwise.
-
-        Args:
-            files_to_sync: A list of SyncFilesItem.
-            full_version: Version to be set in the spec-file.
-            upstream_tag: The commit message of this commit is going to be used
-                to update the changelog in the spec-file.
-            resolved_bugs: List of bugs that are resolved by the update (e.g. [rhbz#123]).
-
-        Returns:
-            The list of synced files with the spec-file removed if it was updated.
+        If a SyncFilesItem represents a directory (whose src points to a directory)
+        that might contain the spec file, an extra rsync filter is added to exclude
+        the spec file (using its basename). This ensures that the upstream spec file
+        is never synced.
         """
         if self.package_config.sync_changelog:
             return files_to_sync
 
-        # add entry to changelog
+        # add entry to changelog even though the spec file is not synced
         ChangelogHelper(self.up, self.dg, self.package_config).update_dist_git(
             full_version=full_version,
             upstream_tag=upstream_tag,
             resolved_bugs=resolved_bugs,
         )
 
-        # exclude spec, we have special plans for it
-        return list(
-            filter(
-                None,
-                [
-                    x.drop_src(self.up.get_absolute_specfile_path())
-                    for x in files_to_sync
-                ],
-            ),
-        )
+        specfile = self.up.get_absolute_specfile_path()
+        spec_basename = os.path.basename(specfile)
```

Comment:
Why basename? Doesn't it make the filter potentially ambiguous?

Reply:
Indeed. I was hoping that rsync would be clever enough to accept absolute paths, but alas it doesn't work. Apparently the absolute paths have a similar logic to gitignore.

One more elegant approach is to first check if it is a relative path in the src tree.
```python
for src in filtered_item.src:
    if specfile.is_relative_to(src):
        ...
```